task_2/src/controllers/authController.ts

```diff
@@ -0,0 +1,115 @@
+import { Request, Response } from 'express'
+import jwt from 'jsonwebtoken'
+import { loginSchema } from '../models/schema/loginSchema.js'
+import { AuthService } from '../services/Auth.js'
+import { UserService } from '../services/User.js'
+import {
+  ALREADY_LOGGED_OUT,
+  JWT_COOKIE_IS_CLEARED,
+  JWT_COOKIE_NOT_PROVIDED,
+  ACCESS_TOKEN_UPDATED,
+} from '../services/constants.js'
+import { getGenericErrorMessage, logError } from '../utils/index.js'
+
+const authService = new AuthService()
+
+export const login = async (req: Request, res: Response) => {
+  const schemaValidation = loginSchema.validate(req.body)
+
+  if (schemaValidation.error) {
+    return res.status(400).json({
+      success: false,
+      message: schemaValidation.error.details[0].message,
+      error: schemaValidation.error,
+    })
+  }
+
+  const { username, password } = schemaValidation.value
+
+  try {
+    const result = await authService.login(username, password)
+
+    if (result?.error) {
+      return res.status(result.status).json(result.error)
+    }
+
+    const {
+      success,
+      message,
+      data: { accessToken, refreshToken },
+    } = result.response
+
+    res.cookie('jwt', refreshToken, { httpOnly: true, sameSite: 'none', secure: true, maxAge: 24 * 60 * 60 * 1000 })
```

Comment:
accourding to standards, usually token is sent inside header like `Authorization: ${jwtToken}`

Reply:
also it would be just a little bit performant if you use already calculated value for maxAge

Reply:
Updated